test/promisified/admin/fetch_topic_offsets_by_timestamp.spec.js

```diff
@@ -0,0 +1,178 @@
+jest.setTimeout(30000);
+
+const { ErrorCodes } = require("../../../lib").KafkaJS;
+const {
+    secureRandom,
+    createTopic,
+    createProducer,
+    createAdmin,
+} = require("../testhelpers");
+
+describe("fetchTopicOffsetsByTimestamp function", () => {
+    let topicName, admin, producer;
+
+    beforeEach(async () => {
+        admin = createAdmin({});
+        producer = createProducer({
+            clientId: "test-producer-id",
+        });
+
+        await admin.connect();
+        await producer.connect();
+
+        topicName = `test-topic-${secureRandom()}`;
+    });
+
+    afterEach(async () => {
+        await admin.deleteTopics({
+            topics: [topicName],
+        });
+        await admin.disconnect();
+        producer && (await producer.disconnect());
+    });
+
+    it("should timeout when fetching topic offsets by timestamp", async () => {
+        await createTopic({ topic: topicName, partitions: 1 });
+
+        await expect(
+            admin.fetchTopicOffsetsByTimestamp(topicName, Date.now(), { timeout: 0 })
+        ).rejects.toHaveProperty("code", ErrorCodes.ERR__TIMED_OUT);
+    });
+
+    it("should fetch offsets for specific timestamps (t1, t2, t3)", async () => {
+        await createTopic({ topic: topicName, partitions: 1 });
+
+        // Messages with specific timestamps
+        const now = 10000000;
+        const t1 = now + 100;
+        const t2 = now + 250;
+        const t3 = now + 400;
+
+        const messages = [
+            { value: 'message1', timestamp: t1.toString() },
+            { value: 'message2', timestamp: t2.toString() },
+            { value: 'message3', timestamp: t3.toString() },
+        ];
+
+        await producer.send({ topic: topicName, messages });
+
+        const offsetsAtSpecificTimestamp1 = await admin.fetchTopicOffsetsByTimestamp(topicName, now + 50);
+        expect(offsetsAtSpecificTimestamp1).toEqual([
+            {
+                partition: 0,
+                offset: "0",
+            },
+        ]);
+
+        const offsetsAtSpecificTimestamp2 = await admin.fetchTopicOffsetsByTimestamp(topicName, now + 150);
```

Comment:
```suggestion
        const offsetsAtSpecificTimestamp2 = await admin.fetchTopicOffsetsByTimestamp(topicName, now + 250);
```

(tests the 'equal' case)